### PR TITLE
Jhford rewrite jonasfj patches

### PR DESCRIPTION
**Pull request by jonasfj**
A few suggestions... still needs some refactor around `determineSpotBids`; and definitely needs to be able to cancel pending if `change` is negative..

@jhford does this make any sense?

### CODE REVIEW

File: provisioner/data.js [hunk 1]
```diff
@@ -529,6 +521,10 @@ WorkerType.prototype.determineSpotBids = function(regions, pricing, runningCapac
   
   var change = this.determineCapacityChange(runningCapacity, pendingCapacity, pending);
 
+  // TODO: If change is negative we should cancel spot requests... This should
+  // go through the aws-manager. IMO this method belongs in the provision.js
+  // file...
```
Comment (jonasfj) [inline provisioner/data.js]: If we do this... then we're compat with existing provisioner when `scalingRatio = 0` and should roll this out...

I suggest you move `determineSpotBids` into `provsion.js` and add the support for cancelling spot requests.

We may have iterations where we alternate between requesting and canceling spot requests.. Which IMO is fine... We can optimize by prefering nodes that has capacity that isn't above the change we want. Ie. reducing utility by the capacity to desiredChange ratio. But that's not important for now...

Get cancel spot requests done right... this is something we do while running. if `change < 0`.
Then roll this out, it looks really good in general!

Comment (jonasfj) [inline provisioner/data.js]: The old provisioner does it here:
https://github.com/taskcluster/aws-provisioner/blob/master/provisioner/provision.js#L19-L56

Decide how many to cancel and how many to request... The `chance` returned from `determineCapacityChange` in this PR is already corrected for max/min...

**Comment by jonasfj** [comment]
@jhford, didn't copy in most of these changes a long time ago in:
https://github.com/taskcluster/aws-provisioner/commit/735327f94473781116c53dc48712684cbbd894ae

And replated commits?

If so, let's close this PR...

**Comment by jhford** [comment]
Yes, they are in commit 735327f94473781116c53dc48712684cbbd894ae

Closing this PR

**Pull request closed without merging**